Packs/Tenable_sc/Integrations/Tenable_sc/Tenable_sc.py

```diff
@@ -1765,95 +2450,448 @@ def get_all_scan_results_command():
-        list_repositories_command()
-    elif demisto.command() == 'tenable-sc-list-credentials':
-        list_credentials_command()
-    elif demisto.command() == 'tenable-sc-list-zones':
-        list_zones_command()
-    elif demisto.command() == 'tenable-sc-list-report-definitions':
-        list_report_definitions_command()
-    elif demisto.command() == 'tenable-sc-list-assets':
-        list_assets_command()
-    elif demisto.command() == 'tenable-sc-list-plugins':
-        list_plugins_command()
-    elif demisto.command() == 'tenable-sc-get-asset':
-        get_asset_command()
-    elif demisto.command() == 'tenable-sc-create-asset':
-        create_asset_command()
-    elif demisto.command() == 'tenable-sc-delete-asset':
-        delete_asset_command()
-    elif demisto.command() == 'tenable-sc-create-scan':
-        create_scan_command()
-    elif demisto.command() == 'tenable-sc-launch-scan':
-        launch_scan_command()
-    elif demisto.command() == 'tenable-sc-get-scan-status':
-        get_scan_status_command()
-    elif demisto.command() == 'tenable-sc-get-scan-report':
-        get_scan_report_command()
-    elif demisto.command() == 'tenable-sc-get-vulnerability':
-        get_vulnerability_command()
-    elif demisto.command() == 'tenable-sc-delete-scan':
-        delete_scan_command()
-    elif demisto.command() == 'tenable-sc-get-device':
-        get_device_command()
-    elif demisto.command() == 'tenable-sc-list-users':
-        list_users_command()
-    elif demisto.command() == 'tenable-sc-list-alerts':
-        list_alerts_command()
-    elif demisto.command() == 'tenable-sc-get-alert':
-        get_alert_command()
-    elif demisto.command() == 'tenable-sc-get-system-information':
-        get_system_information_command()
-    elif demisto.command() == 'tenable-sc-get-system-licensing':
-        get_system_licensing_command()
-    elif demisto.command() == 'tenable-sc-get-all-scan-results':
-        get_all_scan_results_command()
-except Exception as e:
-    LOG(e)
-    LOG.print_log(False)
-    return_error(str(e))
-finally:
-    logout()
+    try:
+        client = Client(
+            verify_ssl=verify_ssl,
+            proxy=proxy,
+            user_name=user_name,
+            password=password,
+            access_key=access_key,
+            secret_key=secret_key,
+            url=url
+        )
+        if command == 'fetch-incidents':
+            first_fetch = params.get('fetch_time').strip()
+            fetch_incidents(client, first_fetch)
+        elif command == 'tenable-sc-launch-scan-report':
+            return_results(launch_scan_report_command(args, client))
+        else:
+            return_results(command_dict[command](client, args))
+    except Exception as e:
+        return_error(
+            f'Failed to execute {command} command. Error: {str(e)}'
+        )
+    finally:
+        if (user_name and password) and not (access_key and secret_key):
```

Comment:
this if should be in the logout command.
Can we make client a context object?
call with Clienth as client

Reply:
not sure I understand the question